### PR TITLE
Timeout test fixes

### PR DESCRIPTION
**Pull request by prashantv**

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -768,13 +768,13 @@ func (c *Connection) checkExchanges() {
...
 		if updated == connectionClosed {
-			go func() {
+			go func(connID uint32) {
 				// We cannot close sendCh until we are sure that there are no other goroutines
 				// that may try to write to sendCh.
 				c.inbound.waitForSendCh()
 				c.outbound.waitForSendCh()
 				close(c.sendCh)
-			}()
+			}(c.connID)
```
Comment (akshayjshah) [inline connection.go]: I'm a little confused by this change. Are we just trying to get the connection ID onto the stack for debugging? If that's the case, let's leave a comment.

Comment (prashantv) [inline connection.go]: Yep purely for debugging. Added a comment at the top.

Comment (akshayjshah) [inline connection.go]: :+1:

**Comment by akshayjshah** [comment]
lgtm

**Pull request merged**